cabal-install/cabal-install.cabal

```diff
@@ -38,6 +38,11 @@ Flag lukko
   default:      True
   manual:       True
 
+flag debug-expensive-assertions
+  description: Enable expensive assertions for testing or debugging
+  default:     False
+  manual:      True
```

Comment:
I'm not sure I understand the purpose of this flag. It looks like it is independent of the one in the solver package, so the warning in this PR doesn't indicate whether the slow functions in the solver are enabled. I don't think that there is a way to make the cabal-install and solver packages share one CPP option, so Main.hs would probably need to call [expensiveAssert](https://github.com/haskell/cabal/blob/1da1893e985a5f51e9dae384880ef16e12383a19/cabal-install-solver/src-assertion/Distribution/Client/Utils/Assertion.hs#L17) directly and check for an exception, similarly to how it calls `assert`.

I also think that it would be fine to leave out the warning about expensive assertions, because they are difficult to enable unintentionally. When I added the flag in #4346, I only intended for it to be enabled in one CI job, in order to check the assertions in tests. I think that the expensive assertions are slow enough that developers wouldn't want to enable them regularly when solving for install plans involving Hackage.

Reply:
Oh, the cabal flag must have been split when cabal-install and cabal-install-solver had been split. I guess we (may in the future) use the flag in both, but we should treat them separately, alas.

Reply:
But, as @jneira remarks, using the flag and warning about the flag is a different thing, so I'm OK with simplifying.